src/jenkins/BuildManifest.groovy

```diff
@@ -49,6 +51,23 @@ class BuildManifest implements Serializable {
             this.getArtifactRoot(jobName, buildNumber)
         ].join('/')
     }
+
+    public String getDistUrl(String publicArtifactUrl = 'https://ci.opensearch.org/ci/dbc', String jobName, String buildNumber) {
```

Comment:
This function is used to reverse engineer the url so we can pass in the root path to the test system.  Instead could we update the python code to accept the full path to the build manifest?  It reduces the number of parameters we pass around greatly

Reply:
After looking further, it looks like the root path is used, because both the build manifest (ending in `/builds/opensearch/manifest.yml`) and the bundle manifest (ending in `/dist/opensearch/manifest.yml`) are both used for a few things in the dependency installer - the build manifest is used for downloading maven dependencies, and the bundle manifest is used for downloading the distribution (via `location` param stored in the bundle manifest).

I prefer to keep the root url rather than add parameters here to keep the interface simpler. It's already a bit complex when running Dashboards tests since multiple root urls are needed already. Lmk what you think.

Reply:
Also note that `getDistUrl()` in particular is only used in the distribution-build workflow to generate the build manifest url to pass to the integ-test workflow. The existing `getArtifactRootUrl()` is the one used for reverse engineering

Reply:
We should consider taking build output such as maven and promoting it to the distribution manifest so we can use a single manifest as input.

Reply:
Wouldn't this be blending the purposes of these manifests too much? From my understanding, the build manifest is for listing out the artifacts and related dependencies per component that are included in the build. The bundle/distribution manifest is a simplified list of built components, with references to their build locations.

Agreed there is quite a bit of overlap between the two though, where maybe it's worth considering merging these into a single manifest? I may not have the most context for why it was designed this way to begin with, but I feel that only one may be needed.

Reply:
Agreed. It's beyond the scope of this issue, but the idea was that the same build is being used to assemble a different distribution, e.g. cross-platform.